interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,189 @@
+
+# Allow querying AppArmor
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionAppArmorSecurityContext"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=session
+    name="core.ubuntu.media.Service",
+
+# Allow communications with unconfined processes
+dbus (receive, send)
+    bus=session
+    path=/com/ubuntu/media/Service{,/**}
+    interface=org.freedesktop.DBus{,.*}
+    peer=(label=unconfined),
+
+# Allow unconfined processes to introspect us
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=unconfined),
+
+# Allow sending mpris signals for session path
+# TODO: We send and receive TrackListReset, research why
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service/sessions/**
+    interface="org.mpris.MediaPlayer2{,.Player,.TrackList}",
+
+# Allow sending properties signals for session path
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service/sessions/**
+    interface="org.freedesktop.DBus.Properties",
+`
+
+const mediaHubConnectedSlotAppArmor = `
+# Allow clients to query/modify and get notified of service properties
+dbus (receive, send)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow client to introspect our DBus api
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    path=/core/ubuntu/media/Service
+    member="Introspect"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to manage Player sessions
+dbus (receive)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const mediaHubConnectedPlugAppArmor = `
+# Description: Allow using media-hub service.
+
+#include <abstractions/dbus-session-strict>
+
+# Allow clients to query/modify and get notified of service properties
+dbus (receive, send)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Allow client to introspect our DBus api
+dbus (send)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    path=/core/ubuntu/media/Service
+    member="Introspect"
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Allow clients to manage Player sessions
+dbus (send)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+`
+
+const mediaHubPermanentSlotSecComp = `
+# Description: Allow operating as the media-hub service. This gives
+# privileged access to the system.
```

Comment:
'This gives privileged access to the system can be dropped'. The 'bind' here doesn't give that.